src/Graphs/partitionedgraphs/partitionedgraph.jl

```diff
@@ -0,0 +1,167 @@
+struct PartitionedGraph{V,PV,G<:AbstractNamedGraph{V},PG<:AbstractNamedGraph{PV}} <:
+       AbstractPartitionedGraph{V,PV}
+  graph::G
+  partitioned_graph::PG
+  partitioned_vertices::Dictionary{PV,Vector{V}}
+  which_partition::Dictionary{V,PV}
+end
+
+##Constructors.
+function PartitionedGraph{V,PV,G,PG}(
+  g::AbstractNamedGraph{V}, partitioned_vertices::Dictionary{PV,Vector{V}}
+) where {V,PV,G<:AbstractNamedGraph{V},PG<:AbstractNamedGraph{PV}}
+  pvs = keys(partitioned_vertices)
+  pg = NamedGraph(pvs)
+  which_partition = Dictionary{V,PV}()
+  for v in vertices(g)
+    v_pvs = Set(findall(pv -> v ∈ partitioned_vertices[pv], pvs))
+    @assert length(v_pvs) == 1
+    insert!(which_partition, v, first(v_pvs))
+  end
+
+  for e in edges(g)
+    pv_src, pv_dst = which_partition[src(e)], which_partition[dst(e)]
+    pe = NamedEdge(pv_src => pv_dst)
+    if pv_src != pv_dst && !has_edge(pg, pe)
+      add_edge!(pg, pe)
+    end
+  end
+
+  return PartitionedGraph(g, pg, partitioned_vertices, which_partition)
+end
+
+function PartitionedGraph(
+  g::AbstractNamedGraph{V}, partitioned_vertices::Dictionary{PV,Vector{V}}
+) where {V,PV}
+  return PartitionedGraph{V,PV,NamedGraph{V},NamedGraph{PV}}(g, partitioned_vertices)
+end
+
+function PartitionedGraph{V,Int64,G,NamedGraph{Int64}}(
+  g::AbstractNamedGraph{V}, partitioned_vertices::Vector{Vector{V}}
+) where {V,G<:NamedGraph{V}}
+  partitioned_vertices_dict = Dictionary{Int64,Vector{V}}(
+    [i for i in 1:length(partitioned_vertices)], partitioned_vertices
+  )
+  return PartitionedGraph{V,Int64,NamedGraph{V},NamedGraph{Int64}}(
+    g, partitioned_vertices_dict
+  )
+end
+
+function PartitionedGraph(
+  g::AbstractNamedGraph{V}, partition_vertices::Vector{Vector{V}}
+) where {V}
+  return PartitionedGraph{V,Int64,NamedGraph{V},NamedGraph{Int64}}(g, partition_vertices)
+end
+
+function PartitionedGraph{V,V,G,G}(vertices::Vector{V}) where {V,G<:NamedGraph{V}}
+  return PartitionedGraph(NamedGraph{V}(vertices), [[v] for v in vertices])
+end
+
+function PartitionedGraph(vertices::Vector{V}) where {V}
+  return PartitionedGraph{V,V,NamedGraph{V},NamedGraph{V}}(vertices)
+end
+
+function PartitionedGraph(
+  g::AbstractNamedGraph{V};
```

Comment:
```suggestion
  g::AbstractNamedGraph;
```
`V` doesn't seem to be used.